### PR TITLE
Core 47229 - added identityMap as part of xdm obj

### PR DESCRIPTION
**Pull request by ninaceban**
## Description

Added identityMap as part of xdm object and it's specific validation.
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [ ] I have made any necessary test changes and all tests pass.
- [ ] I have run the Sandbox successfully.

### CODE REVIEW

File: src/components/DataCollector/validateUserEventOptions.js [hunk 1]
```diff
@@ -22,7 +33,20 @@ export default ({ options, logger }) => {
...
+            id: string().required(),
+            primary: boolean().default(false),
+            authenticatedState: enumOf(
+              AMBIGUOUS,
+              AUTHENTICATED,
+              LOGGED_OUT
+            ).default(AMBIGUOUS)
+          }).noUnknownFields()
+        )
+      )
```
Comment (jonsnyder) [inline src/components/DataCollector/validateUserEventOptions.js:49]: 👍

File: src/components/DataCollector/validateUserEventOptions.js [hunk 2]
```diff
@@ -22,7 +33,20 @@ export default ({ options, logger }) => {
   const eventOptionsValidator = objectOf({
     type: string(),
     xdm: objectOf({
-      eventType: string()
+      eventType: string(),
+      identityMap: mapOfValues(
+        arrayOf(
+          objectOf({
+            id: string().required(),
```
Comment (liljenback) [inline src/components/DataCollector/validateUserEventOptions.js:40]: ID is not listed as required in the XDM schema

File: src/components/DataCollector/validateUserEventOptions.js [hunk 3]
```diff
@@ -22,7 +33,20 @@ export default ({ options, logger }) => {
...
+            primary: boolean().default(false),
+            authenticatedState: enumOf(
+              AMBIGUOUS,
+              AUTHENTICATED,
+              LOGGED_OUT
+            ).default(AMBIGUOUS)
+          }).noUnknownFields()
```
Comment (liljenback) [inline src/components/DataCollector/validateUserEventOptions.js:47]: There can be unknown fields here (based on the schema)

File: src/components/DataCollector/validateUserEventOptions.js (hunk 2 above)
Comment (ninaceban) [reply src/components/DataCollector/validateUserEventOptions.js:40]: ok, will relax the validation a little bit

**Pull request merged**